addons/microsoft_calendar/models/res_users.py

```diff
@@ -139,12 +152,22 @@ def _sync_all_microsoft_calendar(self):
                 _logger.exception("[%s] Calendar Synchro - Exception : %s!", user, exception_to_unicode(e))
                 self.env.cr.rollback()
 
+    def _check_microsoft_calendar_credentials(self):
+        # Create the Calendar Credentials for the current user if not already created.
+        if not self.microsoft_calendar_account_id:
+            self.microsoft_calendar_account_id = self.env['microsoft.calendar.credentials'].sudo().create([{'user_ids': [Command.set(self.ids)]}])
```

Comment:
This won't work if `len(self.ids) > 1` (credentials cannot be shared with multiple users)